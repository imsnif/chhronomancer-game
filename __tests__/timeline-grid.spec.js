import 'react-native'
import React from 'react'
import TimelineGrid from '../components/timeline-grid'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('TimelineGrid (one timeline) => renders correctly', () => {
  const timeline1 = {
    name: 'Timeline 1',
    type: 'steal',
    players: [1, 2]
  }
  const getTimeline = jest.fn().mockReturnValue(timeline1)
  const playersStore = {
    players: [
      {id: 1, name: 'foo', items: {}},
      {id: 2, name: 'bar', items: {}}
    ]
  }
  const tree = renderer.create(
    <TimelineGrid
      playersStore={playersStore}
      timelineStore={{getTimeline, timelines: [timeline1]}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('TimelineGrid (multiple timelines) => renders correctly', () => {
  const timeline1 = {
    name: 'Timeline 1',
    type: 'steal',
    players: [1, 2]
  }
  const timeline2 = {
    name: 'Timeline 2',
    type: 'assistPrevent',
    players: [1]
  }
  const getTimeline = jest.fn()
    .mockReturnValueOnce(timeline1)
    .mockReturnValueOnce(timeline2)
  const playersStore = {
    players: [
      {id: 1, name: 'foo', items: {}},
      {id: 2, name: 'bar', items: {}}
    ]
  }
  const tree = renderer.create(
    <TimelineGrid
      playersStore={playersStore}
      timelineStore={{getTimeline, timelines: [timeline1, timeline2]}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('TimelineGrid (no timelines) => renders correctly', () => {
  const getTimeline = jest.fn().mockReturnValue({players: []})
  const playersStore = {
    players: [
      {id: 1, name: 'foo', items: {}},
      {id: 2, name: 'bar', items: {}}
    ]
  }
  const tree = renderer.create(
    <TimelineGrid
      playersStore={playersStore}
      timelineStore={{timelines: [], getTimeline}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
