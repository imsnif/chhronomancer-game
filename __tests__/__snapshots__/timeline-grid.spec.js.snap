// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineGrid => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#141313",
      "flexDirection": "column",
      "height": "100%",
      "paddingBottom": 5,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#329932",
                  "borderColor": "#329932",
                  "borderRadius": 0,
                  "borderWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 120,
                  "marginBottom": 7,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "backgroundColor": "#141313",
                      "borderColor": "black",
                      "borderWidth": 1,
                      "flex": 0,
                      "height": 32,
                      "margin": 1,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#0e0e0e",
                      "flex": 7,
                      "fontFamily": "telegrama_raw",
                      "fontSize": 30.607554623001164,
                      "textAlign": "center",
                    }
                  }
                >
                  Timeline 1
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#141313",
                    "borderColor": "#141313",
                    "borderRadius": 0,
                    "borderWidth": 0.5,
                    "flex": 13,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "margin": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#141313",
                  "flex": 0,
                  "height": 120,
                  "marginBottom": 7,
                  "opacity": 0.7,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#329932",
                    "fontFamily": "telegrama_raw",
                    "fontSize": 153.03777311500582,
                    "height": "100%",
                    "textAlignVertical": "center",
                  }
                }
              >
                LOCKED
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#329932",
                  "borderColor": "#329932",
                  "borderRadius": 0,
                  "borderWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 120,
                  "marginBottom": 7,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "backgroundColor": "#141313",
                      "borderColor": "black",
                      "borderWidth": 1,
                      "flex": 0,
                      "height": 32,
                      "margin": 1,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#0e0e0e",
                      "flex": 7,
                      "fontFamily": "telegrama_raw",
                      "fontSize": 30.607554623001164,
                      "textAlign": "center",
                    }
                  }
                >
                  Timeline 2
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#141313",
                    "borderColor": "#141313",
                    "borderRadius": 0,
                    "borderWidth": 0.5,
                    "flex": 13,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "margin": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#329932",
                  "borderColor": "#329932",
                  "borderRadius": 0,
                  "borderWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 120,
                  "marginBottom": 7,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "backgroundColor": "#141313",
                      "borderColor": "black",
                      "borderWidth": 1,
                      "flex": 0,
                      "height": 32,
                      "margin": 1,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#0e0e0e",
                      "flex": 7,
                      "fontFamily": "telegrama_raw",
                      "fontSize": 30.607554623001164,
                      "textAlign": "center",
                    }
                  }
                >
                  Timeline 3
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#141313",
                    "borderColor": "#141313",
                    "borderRadius": 0,
                    "borderWidth": 0.5,
                    "flex": 13,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "margin": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                  <Image
                    source={
                      Object {
                        "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
                      }
                    }
                    style={
                      Object {
                        "height": 30,
                        "margin": 4,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "green",
                          "height": "100%",
                          "opacity": 0.5,
                          "width": "100%",
                        }
                      }
                    />
                  </Image>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#329932",
                  "borderColor": "#329932",
                  "borderRadius": 0,
                  "borderWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 120,
                  "marginBottom": 7,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 35,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "backgroundColor": "#141313",
                      "borderColor": "black",
                      "borderWidth": 1,
                      "flex": 0,
                      "height": 32,
                      "margin": 1,
                      "resizeMode": "contain",
                      "width": 32,
                    }
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#0e0e0e",
                      "flex": 7,
                      "fontFamily": "telegrama_raw",
                      "fontSize": 30.607554623001164,
                      "textAlign": "center",
                    }
                  }
                >
                  Timeline 4
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#141313",
                    "borderColor": "#141313",
                    "borderRadius": 0,
                    "borderWidth": 0.5,
                    "flex": 13,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "margin": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": "20%",
          "resizeMode": "stretch",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
            "marginTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#329932",
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "backgroundColor": "#329932",
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#0e0e0e",
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              all timelines
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "backgroundColor": "#141313",
                  "borderBottomColor": "#329932",
                  "borderBottomWidth": 1,
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#329932",
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              my timelines
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 4,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#329932",
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "backgroundColor": "#329932",
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#0e0e0e",
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              name
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "backgroundColor": "#141313",
                  "borderBottomColor": "#329932",
                  "borderBottomWidth": 1,
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#329932",
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              type
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
